File: Frontend/clinicflow-web/src/api/appointments.ts
import axios from "axios";
import { api } from "./axios";

export async function getAppointments() {
  const res = await api.get("/api/Appointments");
  return res.data;
}

export async function createAppointment(dto: any) {
  const res = await api.post("/api/Appointments", dto);
  return res.data;
}

export async function updateAppointment(id: number, status: string) {
  const res = await api.patch(`/api/Appointments/${id}/status`, { status });
  return res.data;
}

export async function deleteAppointment(id: number) {
  await api.delete(`/api/Appointments/${id}`);
}

export interface Appointment {
  id: number;
  patientName: string;
  doctorId: number;
  doctorName: string;
  startTime: string;
  endTime: string;
  status: string;
}

export interface CreateAppointmentDto {
  patientName: string;
  doctorId: number | null;
  startTime: string;
  endTime: string;
  status: string;
}

export async function getDoctorBookedSlots(doctorId: number, date: string) {
  // date format expected: YYYY-MM-DD
  const res = await api.get(`/api/Appointments/doctor/${doctorId}/date/${date}`);
  return res.data; // array of { startTime, endTime }
}

File: Backend/ClinicFlow/ClinicFlow.Api/Models/Appointments.cs
using System.ComponentModel.DataAnnotations;

namespace ClinicFlow.Api.Models;

public class Appointment
{
    public int Id { get; set; }

    [Required]
    [MaxLength(100)]
    public string PatientName { get; set; } = string.Empty;

    [Required]
    [MaxLength(100)]
    public string DoctorName { get; set; } = string.Empty;

    [Required]
    public DateTime StartTime { get; set; }

    [Required]
    public DateTime EndTime { get; set; }

    [Required]
    [MaxLength(30)]
    public string Status { get; set; } = "Scheduled";
}
